marimo/_plugins/ui/_impl/tables/ibis_table.py

```diff
@@ -0,0 +1,219 @@
+
+from typing import Any, Optional, Tuple
+
+from marimo._data.models import (
+    ColumnSummary,
+    ExternalDataType,
+)
+from marimo._dependencies.dependencies import DependencyManager
+from marimo._plugins.ui._impl.tables.format import (
+    FormatMapping,
+)
+from marimo._plugins.ui._impl.tables.pandas_table import (
+    PandasTableManagerFactory,
+)
+from marimo._plugins.ui._impl.tables.polars_table import (
+    PolarsTableManagerFactory,
+)
+from marimo._plugins.ui._impl.tables.pyarrow_table import (
+    PyArrowTableManagerFactory,
+)
+from marimo._plugins.ui._impl.tables.table_manager import (
+    ColumnName,
+    FieldType,
+    FieldTypes,
+    TableManager,
+    TableManagerFactory,
+)
+from marimo._utils.memoize import memoize_last_value
+
+
+class IbisTableManagerFactory(TableManagerFactory):
+    @staticmethod
+    def package_name() -> str:
+        return "ibis"
+
+    @staticmethod
+    def create() -> type[TableManager[Any]]:
+        import ibis  # type: ignore
+
+        class IbisTableManager(TableManager[ibis.Table]):
+            type = "ibis"
+
+            def to_csv(
+                self, format_mapping: Optional[FormatMapping] = None
+            ) -> bytes:
+                return self._as_table_manager().to_csv(format_mapping)
+
+            def to_json(self) -> bytes:
+                return self._as_table_manager().to_json()
+
+            def supports_download(self) -> bool:
+                return False
+
+            def apply_formatting(
+                self, format_mapping: FormatMapping
+            ) -> ibis.Table:
+                raise NotImplementedError("Column formatting not supported")
+
+            def supports_filters(self) -> bool:
+                return True
+
+            def select_rows(
+                self, indices: list[int]
+            ) -> TableManager[ibis.Table]:
+                if not indices:
+                    return self.take(0, 0)  # Return empty table
+                # Select rows using Ibis API
+                return IbisTableManager(
+                    self.data.filter(ibis.row_number().over().isin(indices))
+                )
+
+            def select_columns(
+                self, columns: list[str]
+            ) -> TableManager[ibis.Table]:
+                return IbisTableManager(self.data.select(columns))
+
+            def get_row_headers(
+                self,
+            ) -> list[str]:
+                return []
+
+            @staticmethod
+            def is_type(value: Any) -> bool:
+                return isinstance(value, ibis.Table)
+
+            def get_field_types(self) -> FieldTypes:
+                return {
+                    column: IbisTableManager._get_field_type(self.data[column])
+                    for column in self.data.columns
+                }
+
+            def take(self, count: int, offset: int) -> IbisTableManager:
+                if count < 0:
+                    raise ValueError("Count must be a positive integer")
```

Comment:
Does count need to be positive (in which case the check should be `if count <=0`) or non-negative (in which case the check is fine)?